pkg/extprom/http/instrument_server.go

```diff
@@ -57,35 +65,57 @@ func NewInstrumentationMiddleware(reg prometheus.Registerer, buckets []float64)
 				Help:    "Tracks the latencies for HTTP requests.",
 				Buckets: buckets,
 			},
-			[]string{"code", "handler", "method"},
+			[]string{"code", "handler", "method", "tenant_id"},
 		),
 
 		requestSize: promauto.With(reg).NewSummaryVec(
 			prometheus.SummaryOpts{
 				Name: "http_request_size_bytes",
 				Help: "Tracks the size of HTTP requests.",
 			},
-			[]string{"code", "handler", "method"},
+			[]string{"code", "handler", "method", "tenant_id"},
 		),
 
 		requestsTotal: promauto.With(reg).NewCounterVec(
 			prometheus.CounterOpts{
 				Name: "http_requests_total",
 				Help: "Tracks the number of HTTP requests.",
-			}, []string{"code", "handler", "method"},
+			}, []string{"code", "handler", "method", "tenant_id"},
 		),
 
 		responseSize: promauto.With(reg).NewSummaryVec(
 			prometheus.SummaryOpts{
 				Name: "http_response_size_bytes",
 				Help: "Tracks the size of HTTP responses.",
 			},
-			[]string{"code", "handler", "method"},
+			[]string{"code", "handler", "method", "tenant_id"},
 		),
 	}
 	return &ins
 }
 
+func getTenantIdentifier(ctx context.Context) string {
+	return ctx.Value("tenantID").(string)
+}
+
+// InsertTenantIdentifier inserts tenant identifier in metrics.
+func (ins *defaultInstrumentationMiddleware) InsertTenantIdentifier(next http.Handler) http.HandlerFunc {
+	return func (w http.ResponseWriter, r *http.Request) {
+		tenantIdentifier := getTenantIdentifier(r.Context())
+		if tenantIdentifier != "" {
+			ins.requestSize.CurryWith(prometheus.Labels{"tenant_id": tenantIdentifier})
```

Comment:
I thought `CurryWith` is actually returning a metric which carries this label. But no one use this?